src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -198,20 +252,43 @@ private static String createUser(final Map<String, String> propertiesReadFromCli
       return user.getValidationErrorMessage();
     }
 
+    if (new DbUserController().doesUserExist(user.getName())) {
+      return "That user name has already been taken";
+    }
+    final String base64 = propertiesReadFromClient.get(ENCRYPTED_PASSWORD_KEY);
+    if (base64 != null) {
+      return decryptPassword(base64, propertiesSentToClient.get(RSA_PUBLIC_KEY), pass -> {
+        new DbUserController().createUser(user, pass);
+        return null;
+      });
+    }
+
     final HashedPassword password = new HashedPassword(propertiesReadFromClient.get(HASHED_PASSWORD_KEY));
     if (!password.isValidSyntax()) {
       return "Password is not hashed correctly";
     }
 
-    if (new DbUserController().doesUserExist(user.getName())) {
-      return "That user name has already been taken";
-    }
-
     try {
       new DbUserController().createUser(user, password);
       return null;
     } catch (final Exception e) {
       return e.getMessage();
     }
   }
+
+  private static String decryptPassword(final String base64, final String publicKey,
+      final Function<String, String> function) {
+    Preconditions.checkNotNull(base64);
+    try {
+      final Cipher cipher = Cipher.getInstance(RSA_ECB_OAEPP);
+      cipher.init(Cipher.DECRYPT_MODE, rsaKeyMap.get(publicKey));
+      return function.apply(new String(cipher.doFinal(Base64.getDecoder().decode(base64)), StandardCharsets.UTF_8));
```

Comment:
You might want to consider extracting a class (e.g. `LobbyAuthenticator`) that can be shared between `LobbyLogin` and `LobbyLoginValidator` that encapsulates all the crypto.  There probably needs to be a `g.s.engine.lobby.common` package for code that can be used by both the client and server.